_posts/2020-10-12-sparklyr-flint-0.2.0-released/sparklyr-flint-0.2.0.Rmd

```diff
@@ -0,0 +1,143 @@
+##  4 1970-01-01 00:00:04     4     5
+##  5 1970-01-01 00:00:05     5     6
+##  6 1970-01-01 00:00:06     6     7
+##  7 1970-01-01 00:00:07     7     8
+##  8 1970-01-01 00:00:08     8     9
+##  9 1970-01-01 00:00:09     9    10
+## 10 1970-01-01 00:00:10    10    11
+```
+
+Notice regardless of which temporal direction is selected, all timestamps and `u` values from `left` will always be present in the output, and the `v` column in the output contains `NA` whenever there is no record from `right` that meets the matching criteria.
+
+## OLS Regression
+
+You might be wondering whether the version of this functionality in Flint is more or less identical to `lm()` in R. Turns out it has much more to offer than `lm()` does. A OLS regression in Flint will compute useful metrics such as [Akaike information criterion](https://en.wikipedia.org/wiki/Akaike_information_criterion) and [Bayesian information criterion](https://en.wikipedia.org/wiki/Bayesian_information_criterion), both of which are useful more model selection purposes, and the calculations of both are parallelized by Flint to fully utilize computational power available in a Spark cluster. In addition, Flint supports ignoring regressors that are constant or nearly constant, which becomes useful when an intercept term is included. To see why this is the case, we need to briefly examine the goal of the OLS regression, which is to find some column vector of coefficients $\mathbf{\beta}$ that minimizes $\|\mathbf{y} - \mathbf{X} \mathbf{\beta}\|^2$, where $\mathbf{y}$ is the column vector of response variables, and $\mathbf{X}$ is a matrix consisting of columns of regressors plus an entire column of $1$s representing the intercept terms. The solution to this problem is $\mathbf{\beta} = (\mathbf{X}^\intercal\mathbf{X})^{-1}\mathbf{X}^\intercal\mathbf{y}$, assuming the Gram matrix $\mathbf{X}^\intercal\mathbf{X}$ is non-singular. However, if $\mathbf{X}$ contains a column of all $1$s of intercept terms, and another column of formed by a regressor that is constant (or so), then columns of $\mathbf{X}$ will be linearly dependent (or nearly so) and $\mathbf{X}^\intercal\mathbf{X}$ will be singular (or nearly so), which presents an issue computation-wise. However, if a regressor is constant, then it essentially plays the same role as the intercept terms do, so simply excluding such constant regressor in $\mathbf{X}$ solves the problem. Also, speaking of inverting the Gram matrix, readers remembering the concept of "condition number" from numerical analysis must be thinking to themsolves how computing $\mathbf{\beta} = (\mathbf{X}^\intercal\mathbf{X})^{-1}\mathbf{X}^\intercal\mathbf{y}$ could be numerically unstable if $\mathbf{X}^\intercal\mathbf{X}$ has a large condition number. This is why Flint also outputs the condition number of the Gram matrix in the OLS regression result, so that one can sanity-check the underlying quadratic minimization problem being solved is well-conditioned.
```

Comment:
```suggestion
You might be wondering whether the version of this functionality in Flint is more or less identical to `lm()` in R. Turns out it has much more to offer than `lm()` does. An OLS regression in Flint will compute useful metrics such as [Akaike information criterion](https://en.wikipedia.org/wiki/Akaike_information_criterion) and [Bayesian information criterion](https://en.wikipedia.org/wiki/Bayesian_information_criterion), both of which are useful for model selection purposes, and the calculations of both are parallelized by Flint to fully utilize computational power available in a Spark cluster. In addition, Flint supports ignoring regressors that are constant or nearly constant, which becomes useful when an intercept term is included. To see why this is the case, we need to briefly examine the goal of the OLS regression, which is to find some column vector of coefficients $\mathbf{\beta}$ that minimizes $\|\mathbf{y} - \mathbf{X} \mathbf{\beta}\|^2$, where $\mathbf{y}$ is the column vector of response variables, and $\mathbf{X}$ is a matrix consisting of columns of regressors plus an entire column of $1$s representing the intercept terms. The solution to this problem is $\mathbf{\beta} = (\mathbf{X}^\intercal\mathbf{X})^{-1}\mathbf{X}^\intercal\mathbf{y}$, assuming the Gram matrix $\mathbf{X}^\intercal\mathbf{X}$ is non-singular. However, if $\mathbf{X}$ contains a column of all $1$s of intercept terms, and another column of formed by a regressor that is constant (or so), then columns of $\mathbf{X}$ will be linearly dependent (or nearly so) and $\mathbf{X}^\intercal\mathbf{X}$ will be singular (or nearly so), which presents an issue computation-wise. However, if a regressor is constant, then it essentially plays the same role as the intercept terms do. So simply excluding such a constant regressor in $\mathbf{X}$ solves the problem. Also, speaking of inverting the Gram matrix, readers remembering the concept of "condition number" from numerical analysis must be thinking to themselves how computing $\mathbf{\beta} = (\mathbf{X}^\intercal\mathbf{X})^{-1}\mathbf{X}^\intercal\mathbf{y}$ could be numerically unstable if $\mathbf{X}^\intercal\mathbf{X}$ has a large condition number. This is why Flint also outputs the condition number of the Gram matrix in the OLS regression result, so that one can sanity-check the underlying quadratic minimization problem being solved is well-conditioned.
```

Reply:
Yikes, this diff isn't very helpful (I think I added a comma, changed an "a" to "an," and broke up a long sentence.

For future reference, it's helpful to break up the text a little bit more (with new lines) for "review" purposes.